scanners/java/java.go

```diff
@@ -0,0 +1,266 @@
+package java
+
+import (
+	"github.com/bitrise-io/bitrise-init/detectors/gradle"
+	"github.com/bitrise-io/bitrise-init/detectors/maven"
+	"github.com/bitrise-io/bitrise-init/models"
+	"github.com/bitrise-io/bitrise-init/steps"
+	envmanModels "github.com/bitrise-io/envman/v2/models"
+	"github.com/bitrise-io/go-utils/log"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	ProjectType = "java"
+
+	buildToolInputTitle   = "Build tool"
+	buildToolInputSummary = "The build tool used in the project. Supported options: Gradle, Maven."
+	buildToolGradle       = "Gradle"
+	buildToolMaven        = "Maven"
+
+	testWorkflowID = "run_tests"
+
+	gradleConfigName        = "java-gradle-config"
+	defaultGradleConfigName = "default-java-gradle-config"
+	gradlewPathInputEnvKey  = "GRADLEW_PATH"
+	gradlewPathInputTitle   = "The project's Gradle Wrapper script (gradlew) path."
+	gradlewPathInputSummary = "The project's Gradle Wrapper script (gradlew) path."
+
+	mavenConfigName              = "java-maven-config"
+	defaultMavenConfigName       = "default-java-maven-config"
+	mavenWrapperPathInputEnvKey  = "MAVEN_WRAPPER_PATH"
+	mavenWrapperPathInputTitle   = "The project's Maven Wrapper script (mvnw) path."
+	mavenWrapperPathInputSummary = "The project's Maven Wrapper script (mvnw) path."
+	mavenTestScriptTitle         = `Run Maven tests`
+	mavenTestScriptContent       = `#!/usr/bin/env bash
+# fail if any commands fails
+set -e
+# make pipelines' return status equal the last command to exit with a non-zero status, or zero if all commands exit successfully
+set -o pipefail
+# debug log
+set -x
+
+$MAVEN_WRAPPER_PATH test
+`
+)
+
+type Scanner struct {
+	gradleProject *gradle.Project
+	mavenProject  *maven.Project
+}
+
+func NewScanner() *Scanner {
+	return &Scanner{}
+}
+
+func (s *Scanner) Name() string {
+	return ProjectType
+}
+
+func (s *Scanner) DetectPlatform(searchDir string) (bool, error) {
+	log.TInfof("Searching for Gradle project files...")
+
+	gradleProject, err := gradle.ScanProject(searchDir)
+	if err != nil {
+		return false, err
+	}
+
+	log.TDonef("Gradle project found: %v", gradleProject != nil)
+	if gradleProject != nil {
+		s.gradleProject = gradleProject
+		printGradleProject(*gradleProject)
+		return true, nil
+	}
+
+	log.TInfof("Searching for Maven project files...")
+	mavenProject, err := maven.ScanProject(searchDir)
+	if err != nil {
+		return false, err
+	}
+
+	log.TDonef("Maven project found: %v", mavenProject != nil)
+	if mavenProject != nil {
+		s.mavenProject = mavenProject
+		printMavenProject(*mavenProject)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (s *Scanner) ExcludedScannerNames() []string {
+	return []string{}
```

Comment:
Don't we want to exclude the Android scanner here?

Reply:
It isn't needed, because the Java scanner is the last scanner in the available scanner's list: https://github.com/bitrise-io/bitrise-init/pull/287/files#diff-3f9980967933bdddf5b469600cd6e088d8042c92f27963c0eeb68d2c83a78cf2R85.
Exclusion works in a way, that if a scanner detects a project, then it can exclude next scanners.